### DISCUSSION TITLE
Postgres attach maps `JSONB` column to `VARCHAR` - bug or intended behaviour?

### DISCUSSION THREAD

**Discussion by dhirschfeld** [duckdb/duckdb#11165] [Q&A]
As it says in the title, when I attach a postgres database and query a table with a `JSONB` column it gets mapped as a `VARCHAR` column rather than `JSON` - is this a bug?

**Comment by dhirschfeld** [comment]
I have a `postgres` table:
```sql
create table "InputArguments"
(
    pkid             bigint generated always as identity                primary key,
    kwargs           jsonb                                              not null,
    insert_timestamp timestamp with time zone default CURRENT_TIMESTAMP not null
);
```
When queried from `duckdb` it results in:

![image](https://github.com/duckdb/duckdb/assets/881019/b855d2c3-98cf-4fa9-91b3-06609ad52fb0)


Which I then need to cast to `JSON`. I guess it would be nice if it were automatically mapped to the `JSON` type to avoid the manual cast?

**Comment by anayrat** [comment]
Hello,
I have a similar issue. If I understand, Duckdb doesn't have json type. 
I tried to create a table with a `json` column, it still mapped as `varchar`.
For me, it is a big issue, I can't import a table containing json :(

**Reply by Mause** [reply]
The json type is included in the `json` extension: https://duckdb.org/docs/extensions/json
